ocaml-lsp-server/docs/ocamllsp/polaritySearch-spec.md

```diff
@@ -0,0 +1,40 @@
+# Polarity Search Request
+
+## Description
+
+This custom request allows clients to perform a polarity-based search at a specific position within a text document based on finding functions or types that match a specific query pattern.
+
+## Server capability
+
+- property name: `handlePolaritySearch`
+- property type: `boolean`
+
+## Request
+
+```js
+export interface PolaritySearchParams extends TexDocumentPositionParams
+{
+    query: string;
+    limit: int
+}
+```
+- method: `ocamllsp/polaritySearch`
+- params:
+    - `TextDocumentPositionParams`: This is an existing interface that includes:
+            - `TextDocumentIdentifier`: Specifies the document uri for which the request is sent.
+            - `Position`: Specifies the cursor position.
+        More details can be found in the [TextDocumentPositionParams - LSP Specification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentPositionParams).
+    - `query`: The search pattern.
+    - `limit`: The number of results to return
+
+## Response
+
+```js
+result: PolaritySearch | null
+export interface PolaritySearch {
+    type entry = { path : string; desc : string }
```

Comment:
Biksheeding here but `desc` seems not very relevant, what about `type`?

Reply:
I think `type` should be a keyword that can't be used as a record field

Reply:
We use `type` (and `type_` for the OCaml record) in Type Enclosing